be1-go/hub/standard_hub/mod.go

```diff
@@ -151,68 +181,71 @@ func (h *Hub) Receiver() chan<- socket.IncomingMessage {
 	return h.messageChan
 }
 
-// OnSocketClose implements hub.Hub
-func (h *Hub) OnSocketClose() chan<- string {
-	return h.closedSockets
+// AddServerSocket adds a socket to the sockets known by the hub and query a
+// catchup from the new server
+func (h *Hub) AddServerSocket(socket socket.Socket) error {
+	h.serverSockets.Upsert(socket)
+
+	return h.CatchupToServer(socket, rootChannel)
```

Comment:
Returning a function call makes the code harder to follow. Please do it in two steps.